source/migrate-kmongo.txt

```diff
@@ -0,0 +1,518 @@
+Kotlin. It is developed by the MongoDB team.
+
+Although both drivers :ref:`support synchronous and asynchronous operations <kotlin-sync-async-support>`, 
+the examples on this page will use asynchronous coroutine-based operations. 
+
+Connect to MongoDB Cluster
+--------------------------
+
+Both drivers let you connect to and communicate with MongoDB clusters from a 
+Kotlin application.
+
+.. tabs::
+
+   .. tab::
+      :tabid: {+driver-long+}
+
+      To connect to a MongoDB cluster using the MongoDB Kotlin driver: 
+
+      .. code-block:: kotlin
+
+        import com.mongodb.kotlin.client.coroutine.MongoClient
+
+        data class Jedi(val name: String, val age: Int)   
+
+        // Replace the placeholder with your MongoDB deployment's connection string
+        val uri = CONNECTION_STRING_URI_PLACEHOLDER
+
+        val mongoClient = MongoClient.create(uri)
+        
+        val database = mongoClient.getDatabase("test")
+        // Get a collection of documents of type Jedi
+        val collection = database.getCollection<Jedi>("jedi")
+
+      See the :ref:`Connect to MongoDB <connect-to-mongodb>` documentation for more 
+      information.
+      
+   .. tab::
+      :tabid: KMongo
+
+      To connect to a MongoDB cluster using KMongo with coroutines: 
+
+      .. code-block:: kotlin
+
+        import org.litote.kmongo.reactivestreams.*  
+        import org.litote.kmongo.coroutine.* 
+
+        data class Jedi(val name: String, val age: Int)
+
+        // Get new MongoClient instance using coroutine extension
+        val client = KMongo.createClient().coroutine 
+        
+        val database = client.getDatabase("test")
+        // Get a collection of documents of type Jedi 
+        val col = database.getCollection<Jedi>()
+
+      Unlike the MongoDB Kotlin driver, KMongo allows the collection name to be
+      inferred from the data class name.
+
+CRUD and Aggregation
+--------------------
+
+Both drivers provide support for all MongoDB CRUD APIs and aggregation 
+operations.
+
+.. tabs::
+
+   .. tab::
+      :tabid: {+driver-long+}
+
+      The MongoDB Kotlin driver also provides functions for all basic CRUD operations:
+
+      .. code-block:: kotlin
+
+        // Insert a document
+          val jedi =a Jedi("Luke Skywalker", 19)
+          collection.insertOne(jedi)
+
+          // Find a document
+          val luke = collection.find(Jedi::name.name, "Luke Skywalker")
+          val jedis = collection.find(lt(Jedi::age.name, 30)).toList()
+
+          // Update a document
+          val filter = Filters.eq(Jedi::name.name, "Luke Skywalker")
+          val update = Updates.set(Jedi::age.name, 20)
+          collection.updateOne(filter, update)
+
+          // Delete a document
+          val filter = Filters.eq(Jedi::name.name, "Luke Skywalker")
+          collection.deleteOne(filter)
+
+      Aggregation pipelines can be built using the ``aggregate`` method and the 
+      ``pipeline`` function: 
```

Comment:
```suggestion
      You can build aggregation pipelines by using the ``aggregate`` method and the 
      ``pipeline`` function: 
```

Reply:
S: In general, change from passive to active second person where possible

Reply:
Updated